Makefile

```diff
@@ -276,6 +278,7 @@ coverage-report-js:
 # Runs the C++ tests using the built `cctest` executable.
 cctest: all
 	@out/$(BUILDTYPE)/$@ --gtest_filter=$(GTEST_FILTER)
+	@out/$(BUILDTYPE)/embedtest "require('./test/embedding/test.js')"
```

Comment:
> I think it should work if you just use the addon test config?

@joyeecheung I’m not sure what this refers to … this test is run very differently from the addons?

> This is currently bypassing the `tools/test.py` machinery, so it does not play along too well with the CI which relies on tap output

So do the other cctests, too, if I’m not mistaken?

Reply:
On the CI we use `gtest`'s XML output for `cctest` (gtest dropped tap support some time ago so we just upload the xml to Jenkins at the end of the test jobs):
https://github.com/nodejs/node/blob/f46c44c3578a08bf131524155668719478cfdd35/Makefile#L531

Reply:
@richardlau Any suggestions here? I don’t think we can really implement full TAP support for this test, but if it’s about failing/passing, I guess we could wrap the call to the embedding test in a shell script … I don’t really know what kind of integration that would require

(That being said, I don’t expect this test to be particularly platform-specific, so it’s most likely going to fail locally before we run into trouble in CI)

Reply:
> I’m not sure what this refers to … this test is run very differently from the addons?

The addon tests work by:

1. Building a binary
2. Load that binary in a JS test

I think it's not too different from what's being done here other than that in this case the binary is an executable instead of a shared object (if you are using node.gyp to build it the dependency is handled just fine without extra modification), and we need to load the binary by spawning a process instead of dlopen, both are controllable via configs / source code.

If you use `AddonTestConfiguration` in the `test/folder/testcfg.py`, it would scan all files ending in `.js` in `folder` and run it with `out/${BUILDTYPE}/node(.exe)` with appropriate command line flags, whereas if you use `SimpleTestCase` it needs the tests to be named like `r'^test-.*\.m?js$'`, either is fine, depending on how you want the directory structure to be. Of course the ideal solution would be to add another config in `test/__init__.py` and override e.g. `GetCommand()` to locate `out/${BUILDTYPE}/embedtest(.exe)` and run the test directly to save an extra indirection, but that's not a big deal.

Reply:
(BTW I just noticed that the test is not run on Windows if you just use Makefile to run it)

Reply:
@joyeecheung I understand how this would work *conceptually*, but that doesn’t mean that I know the concrete steps that would be necessary to implement this; in particular:

> if you are using node.gyp to build it the dependency is handled just fine without extra modification

That would mean rebuilding all of Node.js, wouldn’t it?

Either way, if there’s an easy solution, I’m happy to see that implemented, but if this is a blocker for you then I’d prefer opening a separate PR for the test so that this PR can be merged.

Reply:
> That would mean rebuilding all of Node.js, wouldn’t it?

I am talking about what's currently done in the PR is fine (i.e. you don't need extra config in vcbuild.bat to handle the dependency)

This should work though I have not tested it on Windows

```diff
diff --git a/test/embedding/test-embedding.js b/test/embedding/test-embedding.js
new file mode 100644
index 0000000000..6378215ca5
--- /dev/null
+++ b/test/embedding/test-embedding.js
@@ -0,0 +1,25 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const child_process = require('child_process');
+const path = require('path');
+
+common.allowGlobals(global.require);
+let binary = process.features.debug ? `out/Debug/embedtest` : `out/Release/embedtest`;
+if (common.isWindows) {
+  binary += '.exe';
+}
+binary = path.resolve(__dirname, '..', '..', binary);
+
+assert.strictEqual(
+  child_process.spawnSync(binary, ['console.log(42)'])
+    .stdout.toString().trim(),
+  '42');
+
+assert.strictEqual(
+  child_process.spawnSync(binary, ['throw new Error()']).status,
+  1);
+
+assert.strictEqual(
+  child_process.spawnSync(binary, ['process.exitCode = 8']).status,
+  8);
diff --git a/test/embedding/test.js b/test/embedding/test.js
deleted file mode 100644
index a802de1849..0000000000
--- a/test/embedding/test.js
+++ /dev/null
@@ -1,19 +0,0 @@
-'use strict';
-const common = require('../common');
-const assert = require('assert');
-const child_process = require('child_process');
-
-common.allowGlobals(global.require);
-
-assert.strictEqual(
-  child_process.spawnSync(process.execPath, ['console.log(42)'])
-    .stdout.toString().trim(),
-  '42');
-
-assert.strictEqual(
-  child_process.spawnSync(process.execPath, ['throw new Error()']).status,
-  1);
-
-assert.strictEqual(
-  child_process.spawnSync(process.execPath, ['process.exitCode = 8']).status,
-  8);
diff --git a/test/embedding/testcfg.py b/test/embedding/testcfg.py
new file mode 100644
index 0000000000..2c2929f610
--- /dev/null
+++ b/test/embedding/testcfg.py
@@ -0,0 +1,6 @@
+import sys, os
+sys.path.append(os.path.join(os.path.dirname(__file__), '..'))
+import testpy
+
+def GetConfiguration(context, root):
+  return testpy.SimpleTestConfiguration(context, root, 'benchmark')
diff --git a/tools/test.py b/tools/test.py
index 112d5edaff..5eb1fa7672 100755
--- a/tools/test.py
+++ b/tools/test.py
@@ -1491,6 +1491,7 @@ IGNORED_SUITES = [
   'addons',
   'benchmark',
   'doctool',
+  'embedding',
   'internet',
   'js-native-api',
   'node-api',
```

Reply:
(Note: you'd need to add an extra case e.g. read from fixtures for testing the module system as in the diff above the file is directly run with the node binary)

Reply:
@joyeecheung Done, added your changes + fixed up Makefile + an extra test that reads from fixtures in 330f4a78feed0ad16cfe274be70cdaf017104f1f – I’ll kick off CI and we’ll see how that goes.